src/pyhf/constraints.py

```diff
@@ -116,7 +111,19 @@ def logpdf(self, auxdata, pars):
 
         result = prob.Independent(
             prob.Normal(normal_means, self.sigmas), batch_size=self.batch_size
-        ).log_prob(self._dataprojection(auxdata))
+        )
+        return result
+
+    def logpdf(self, auxdata, pars):
+        tensorlib, _ = get_backend()
+        pdf = self.make_pdf(pars)
+        if pdf is None:
+            return (
+                tensorlib.zeros(self.batch_size)
+                if self.batch_size is not None
+                else tensorlib.astensor(0.0)[0]
+            )
+        result = pdf.log_prob(self._dataprojection(auxdata))
```

Comment:
Maybe just return here?

```suggestion
        return pdf.log_prob(self._dataprojection(auxdata))
```